packages/java/src/browser/java-preferences.ts

```diff
@@ -0,0 +1,246 @@
+/********************************************************************************
```

Comment:
They don't have copyrights in the file. I am not sure what I should copy. `vscode-java` is also EPL licence, is it Eclipse project?

// cc @marcdumais

Reply:
`vscode-java` doesn't seem to be an Eclipse Foundation project, since it's not in a Eclipse-controlled GH org. Maybe it's a fork or they re-used some content and kept the original license. I think this borrowing of code requires a CQ (it would not, were the upstream project under the Eclipse Foundation). 

For the attribution aspect, this is a bit different compared to the most usual case we have, where we borrow MIT-licensed code. What we often do then is copy the original license (or add it ourselves if not originally present), but release the whole (original code + our modifications) under our own license (EPLv2 + GPL sub-license). This is permitted because MIT permits sub-licensing.

But since the copied code here is under the EPL v1, I think we can't sub-license the borrowed code, to release it under our own license; I think we would need to release this code under the original license. How to best do that? I would just keep the whole file under EPLv1. You / TypeFox would own the copyright of the changes, but it would be under the original license. Makes sense? We can confirm with the Foundation if there is doubt.

Reply:
@akosyakov @marcdumais _or_ you could ask the folks maintaining vscode-java to help you with this licensing issue. What you most certainly _cannot_ do is copy text mostly verbatim and act like it's new. Btw: the fact that there is no copyright notice in a file does not imply that it is not copyrighted.

Reply:
> What you most certainly cannot do is copy text mostly verbatim and act like it's new. 
[...] the fact that there is no copyright notice in a file does not imply that it is not copyrighted.

Correct for both. The EPL requires we copy/maintain any existing copyright headers. Here there are none, in the file we copied. To make it clear the copied file is kept under EPLv1, I think we should add one ourselves, crediting the original project, as well as a link to the original project/file.

> you could ask the folks maintaining vscode-java to help you with this licensing issue

AFAIK the license of the upstream project and copied file is not in doubt (EPLv1). We could ask their opinion about how to best credit them, but as long as we respect/maintain the original license, I think it's probably not necessary, in this case.